src/native_main.nim

```diff
@@ -217,7 +222,25 @@ proc handleMessage(msg: MessageRecv): string =
             reply.sep = some $DirSep
 
         of "win_firefox_restart":
-            write(stderr, "TODO: NOT IMPLEMENTED\n")
+            when defined windows:
```

Comment:
It is probably worth sketching broadly how this works in a comment - the fact that Firefox waits for this process to exit and then quits itself is a bit magic and not at all obvious from this code alone.